staging/src/k8s.io/apiserver/pkg/admission/plugin/cel/policy_decision.go

```diff
@@ -0,0 +1,53 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cel
+
+import (
+	"encoding/json"
+	"fmt"
+)
+
+type PolicyDecisionKind string
+
+const (
+	Admit PolicyDecisionKind = "Admit"
+	Deny  PolicyDecisionKind = "Deny"
+)
+
+type PolicyDecision struct {
+	Kind    PolicyDecisionKind `json:"kind"`
+	Message any                `json:"message"`
+}
+
+type PolicyDecisionWithMetadata struct {
+	PolicyDecision `json:"decision"`
+	Definition     PolicyDefinition `json:"definition"`
+	Binding        PolicyBinding    `json:"binding"`
+}
+
+type PolicyError struct {
+	Decisions []PolicyDecisionWithMetadata
+}
+
+func (p PolicyError) Error() string {
```

Comment:
`error` type is usually a pointer. not sure if it is better without the pointer receiver.